pkg/menu/menu.go

```diff
@@ -0,0 +1,75 @@
+package menu
+
+import (
+	"io"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+const menuWidth = 50
+
+// newParagraph will return a paragraph object with given initial text.
+func newParagraph(initText string, border bool, location int, wid int, ht int) *widgets.Paragraph {
+	p := widgets.NewParagraph()
+	p.Text = initText
+	p.Border = border
+	p.SetRect(0, location, wid, location+ht)
+	p.TextStyle.Fg = ui.ColorWhite
+	return p
+}
+
+func processInput(input, warning *widgets.Paragraph, isValid func(string) (string, bool), uiEvents <-chan ui.Event) (string, error) {
+	// keep tracking all input from user
+	for {
+		e := <-uiEvents
+		if e.Type != ui.KeyboardEvent {
+			continue
+		}
+		switch e.ID {
+		// if user input control-d, raise error io.EOF meaning no more input.
+		case "<C-d>":
+			return "", io.EOF
+		case "<Enter>":
+			if warningWords, ok := isValid(input.Text); ok {
+				return input.Text, nil
+			} else {
```

Comment:
This is because the warningWords is defined in this if sentence. I will change it to 
warningWords, ok := isValid(input.Text)
if  ok {
	return input.Text, nil
} 
...